backend/src/server_host.py

```diff
@@ -152,6 +157,102 @@ async def get_features(request: Request):
     return await worker.proxy_request(request)
 
 
+def deps_to_dep_info(deps: list[api.Dependency]) -> list[DependencyInfo]:
+    return [
+        DependencyInfo(
+            package_name=dep.pypi_name,
+            display_name=dep.display_name,
+            version=dep.version,
+            extra_index_url=dep.extra_index_url,
+        )
+        for dep in deps
+    ]
+
+
+@app.route("/packages/uninstall", methods=["POST"])
+async def uninstall_dependencies_request(request: Request):
+    full_data = dict(request.json)  # type: ignore
+    package_to_uninstall = full_data["package"]
+
+    packages = await worker.get_packages()
+
+    package = next((x for x in packages if x.name == package_to_uninstall), None)
+
+    if package is None:
+        return json(
+            {"status": "error", "message": f"Package {package_to_uninstall} not found"},
+            status=404,
+        )
+
+    def update_progress(
+        message: str, progress: float, status_progress: float | None = None
+    ):
+        return AppContext.get(request.app).setup_queue.put_and_wait(
+            {
+                "event": "package-install-status",
+                "data": {
+                    "message": message,
+                    "progress": progress,
+                    "statusProgress": status_progress,
+                },
+            },
+            timeout=1,
+        )
+
+    try:
+        await worker.stop()
+        await uninstall_dependencies(
+            deps_to_dep_info(package.dependencies), update_progress, logger
+        )
+        await worker.start()
```

Comment:
This is dangerous. What if the uninstall fails for some reason? We'll have a backend without an active worker.

On that note: how do things like `/nodes` even behave when the worker isn't active? Will they wait forever until something starts the worker, or will they attempt to start the worker if no one is doing so/failed to do so already?

Reply:
I think they'll just wait forever. I'm not sure having multiple things be able to start up the backend separately is a good idea though. i'll fix this however